shell/platform/darwin/macos/framework/Source/FlutterView.mm

```diff
@@ -20,14 +33,73 @@ - (instancetype)initWithFrame:(NSRect)frame
   if (self) {
     self.openGLContext = [[NSOpenGLContext alloc] initWithFormat:shareContext.pixelFormat
                                                     shareContext:shareContext];
+
+    [self setWantsLayer:YES];
+
+    // Layer for content (which will be set by surfaceManager). This is separate from
+    // self.layer because it needs to be flipped vertically (using layer.sublayerTransform)
+    contentLayer = [[CALayer alloc] init];
+    [self.layer addSublayer:contentLayer];
+
+    resizeSynchronizer = [[FlutterResizeSynchronizer alloc] initWithDelegate:self];
+    surfaceManager = [[FlutterSurfaceManager alloc] initWithLayer:contentLayer
+                                                    openGLContext:self.openGLContext];
+
     _reshapeListener = reshapeListener;
-    self.wantsBestResolutionOpenGLSurface = YES;
   }
   return self;
 }
 
+- (void)resizeSynchronizerCommit:(FlutterResizeSynchronizer*)synchronizer {
+  [self.openGLContext makeCurrentContext];
+  glFlush();
+  [NSOpenGLContext clearCurrentContext];
+
+  [CATransaction begin];
+  [CATransaction setDisableActions:YES];
+  self.layer.frame = self.bounds;
+  self.layer.sublayerTransform = CATransform3DTranslate(CATransform3DMakeScale(1, -1, 1), 0,
+                                                        -self.layer.bounds.size.height, 0);
+  contentLayer.frame = self.layer.bounds;
+
+  [surfaceManager swapBuffers];
+
+  [CATransaction commit];
+}
+
+- (int)getFrameBufferIdForSize:(CGSize)size {
+  if ([resizeSynchronizer shouldEnsureSurfaceForSize:size]) {
+    [surfaceManager ensureSurfaceSize:size];
+  }
+  return [surfaceManager glFrameBufferId];
+}
+
+- (void)present {
+  [resizeSynchronizer requestCommit];
+}
+
+- (void)start {
+  active = YES;
+  [self reshaped];
+}
+
+- (void)reshaped {
+  if (active) {
+    CGSize scaledSize = [self convertSizeToBacking:self.bounds.size];
+    [resizeSynchronizer beginResize:scaledSize
+                             notify:^{
```

Comment:
Rather than `notify` what do you think of `onResizeCommit`?

Reply:
Seems confusing. The notify block is basically invoked as soon as we get the mutex (and unblock raster thread). That's at the beginning of resizing. `requestCommit` (which triggers delegate's `resizeSynchronizerCommit`) is when raster thread wants to present frame. 

So presenting frame is called commit here, not sure why starting the resizing process (which notify does) would also be called commit.